rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -194,12 +194,12 @@ rmw_create_client(
 
   auto cleanup_info = rcpputils::make_scope_exit(
     [info, participant_info]() {
-      delete info->pub_listener_;
-      delete info->listener_;
       rmw_fastrtps_shared_cpp::remove_topic_and_type(
-        participant_info, info->response_topic_, info->response_type_support_);
+        participant_info, nullptr, info->response_topic_, info->response_type_support_);
       rmw_fastrtps_shared_cpp::remove_topic_and_type(
-        participant_info, info->request_topic_, info->request_type_support_);
+        participant_info, nullptr, info->request_topic_, info->request_type_support_);
+      delete info->pub_listener_;
+      delete info->listener_;
       delete info;
```

Comment:
Looking at these three consecutive `delete` calls makes me wonder: "Shall we use `unique_ptr` for the listeners so we don't have to worry about their deletion?"

This can be done on a follow-up, though.

Reply:
> Looking at these three consecutive delete calls makes me wonder: "Shall we use unique_ptr for the listeners so we don't have to worry about their deletion?"

Yeah, it's a good question.  The one problem there is that `std::make_unique` doesn't support the `nothrow` variant, which we currently use.  It looks like we could probably define something ourselves (see https://stackoverflow.com/a/57092496), but I think that is probably follow-up work.  So I'm not going to do this here and we can consider it separately.